certbot/main.py

```diff
@@ -653,34 +673,26 @@ def obtain_cert(config, plugins, lineage=None):
         raise
     le_client = _init_le_client(config, auth, installer)
 
-    # SHOWTIME: Possibly obtain/renew a cert, and set action to renew | newcert | reinstall
-    if config.csr is None: # the common case
-        domains, certname = _find_domains_or_certname(config, installer)
-        action, _ = _auth_from_available(le_client, config, domains, certname, lineage)
-    else:
-        assert lineage is None, "Did not expect a CSR with a RenewableCert"
-        _csr_obtain_cert(config, le_client)
-        action = "newcert"
+    if config.csr:
+        cert_path, fullchain_path = _csr_obtain_cert(config, le_client)
+        _report_new_cert(config, cert_path, fullchain_path)
+        _suggest_donation_if_appropriate(config)
+        return
 
-    # POSTPRODUCTION: Cleanup, deployment & reporting
-    notify = zope.component.getUtility(interfaces.IDisplay).notification
-    if config.dry_run:
-        _report_successful_dry_run(config)
-    elif config.verb == "renew":
-        if installer is None:
-            notify("new certificate deployed without reload, fullchain is {0}".format(
-                   lineage.fullchain), pause=False)
-        else:
-            # In case of a renewal, reload server to pick up new certificate.
-            # In principle we could have a configuration option to inhibit this
-            # from happening.
-            installer.restart()
-            notify("new certificate deployed with reload of {0} server; fullchain is {1}".format(
-                   config.installer, lineage.fullchain), pause=False)
-    elif action == "reinstall" and config.verb == "certonly":
+    domains, certname = _find_domains_or_certname(config, installer)
+    should_get_cert, lineage = _find_cert(config, domains, certname)
+
+    if not should_get_cert:
+        notify = zope.component.getUtility(interfaces.IDisplay).notification
         notify("Certificate not yet due for renewal; no action taken.", pause=False)
-    _suggest_donation_if_appropriate(config, action)
+        return
+
+    lineage = _get_and_save_cert(le_client, config, domains, certname, lineage)
 
+    cert_path = lineage.cert_path if lineage else None
```

Comment:
same as above - only if we've done a dry run.

ok, how about I change `_report_new_cert` so that it has a comment about "these will be false if we're doing a dry run", and also change it to `if config.dry_run or not cert_path`.